collectors/sophossiem/test/sophossiem_test.js

```diff
@@ -0,0 +1,234 @@
+const assert = require('assert');
+const sinon = require('sinon');
+var AWS = require('aws-sdk-mock');
+const m_response = require('cfn-response');
+const sophossiemMock = require('./sophossiem_mock');
+var SophossiemCollector = require('../collector').SophossiemCollector;
+const moment = require('moment');
+const utils = require("../utils");
+
+var responseStub = {};
+let getAPILogs;
+describe('Unit Tests', function () {
+
+    beforeEach(function () {
+        AWS.mock('SSM', 'getParameter', function (params, callback) {
+            const data = new Buffer('test-secret');
+            return callback(null, { Parameter: { Value: data.toString('base64') } });
+        });
+        AWS.mock('KMS', 'decrypt', function (params, callback) {
+            const data = {
+                Plaintext: '{}'
+            };
+            return callback(null, data);
+        });
+
+        responseStub = sinon.stub(m_response, 'send').callsFake(
+            function fakeFn(event, mockContext, responseStatus, responseData, physicalResourceId) {
+                mockContext.succeed();
+            });
+    });
+
+    afterEach(function () {
+        responseStub.restore();
+    });
+
+    describe('pawsInitCollectionState', function () {
+        let ctx = {
+            invokedFunctionArn: sophossiemMock.FUNCTION_ARN,
+            fail: function (error) {
+                assert.fail(error);
+            },
+            succeed: function () { }
+        };
+        it('Paws Init Collection State with start date behind 22 hours Success', function (done) {
+            SophossiemCollector.load().then(function (creds) {
+                var collector = new SophossiemCollector(ctx, creds, 'sophossiem');
+                const startDate = moment().subtract(22, 'hours').toISOString();
+                process.env.paws_collection_start_ts = startDate;
+                collector.pawsInitCollectionState({}, (err, initialStates, nextPoll) => {
+                    initialStates.forEach((state) => {
+                        assert.equal(state.poll_interval_sec, 1);
+                        assert.ok(state.from_date);
+                    });
+                    done();
+                });
+            });
+        });
+
+        it('Paws Init Collection State with start date behind 2 days Success', function (done) {
+            SophossiemCollector.load().then(function (creds) {
+                var collector = new SophossiemCollector(ctx, creds, 'sophossiem');
+                const startDate = moment().subtract(2, 'days').toISOString();
+                process.env.paws_collection_start_ts = startDate;
+                collector.pawsInitCollectionState({}, (err, initialStates, nextPoll) => {
+                    initialStates.forEach((state) => {
+                        assert.equal(state.poll_interval_sec, 1);
+                        assert.ok(state.from_date);
```

Comment:
Could we test to make sure that the date is less than 24 hours in the past?

Reply:
will do thanks